core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package main
+
+import (
+    "fmt"
+    "errors"
+    "time"
+    "strconv"
+    "encoding/json"
+
+    "github.com/golang/protobuf/proto"
+    "github.com/hyperledger/fabric-chaincode-go/shim"
+    log "github.com/sirupsen/logrus"
+    "github.com/hyperledger-labs/weaver-dlt-interoperability/core/network/fabric-interop-cc/interfaces/asset-mgmt/protos-go/common"
+)
+
+
+const(
+    defaultTimeLockMillis = 5 * 60 * 1000      // 5 minutes
+)
+
+
+// AssetManagement provides functions to manage asset locks by calling suitable functions in the interop contract
+type AssetManagement struct {
+    shim.Chaincode
+    interopChaincodeId string
+}
+
+
+// Utility functions
+func (am *AssetManagement) Configure(interopChaincodeId string) {
+    am.interopChaincodeId = interopChaincodeId
+}
+
+
+// Ledger transaction (invocation) functions
+
+func (am *AssetManagement) AddFungibleAssetCount(stub shim.ChaincodeStubInterface, assetType string, numUnits int) (bool, error) {
+    var errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if len(assetType) == 0 {
+        errorMsg = "Empty asset type"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if numUnits <= 0 {
+        errorMsg = "Invalid number of asset units"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    iccResp := stub.InvokeChaincode(am.interopChaincodeId, [][]byte{[]byte("AddFungibleAssetCount"), []byte(assetType), []byte(strconv.Itoa(numUnits))}, "")
+    fmt.Printf("Response from Interop CC: %+v\n", iccResp)
+    if iccResp.GetStatus() != shim.OK {
+        return false, fmt.Errorf(string(iccResp.GetPayload()))
+    }
+    fmt.Printf("%d units of asset type %s added to the record\n", numUnits, assetType)
+    return true, nil
+}
+
+func (am *AssetManagement) LockAsset(stub shim.ChaincodeStubInterface, assetAgreement *common.AssetAgreement, lockInfo *common.AssetLock) (bool, error) {
+    var errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if (lockInfo.LockMechanism != common.LockMechanism_HTLC) {
+        log.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+        return false, fmt.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+    }
+    lockInfoHTLC := &common.AssetLockHTLC{}
+    if len(lockInfo.LockInfo) == 0 {
+        log.Error("Empty lock info")
+        return false, fmt.Errorf("Empty lock info")
```

Comment:
for consistency should we introduce a formatted string variable similar to `errorMsg` used in all other instances (e.g. line 75), for the messages that are used across multiple statements? comment applies to the various places a similar pattern is used.

Reply:
Done. Using `errors.New(....)` everywhere now.